packages/pds/src/api/app/bsky/feed/getTimeline.ts

```diff
@@ -21,167 +14,49 @@ export default function (server: Server, ctx: AppContext) {
       const { ref } = db.dynamic
       const requester = auth.credentials.did
 
-      let feedAlgorithm: FeedAlgorithm
-      if (isEnum(FeedAlgorithm, algorithm)) {
-        feedAlgorithm = algorithm
-      } else if (algorithm === undefined) {
-        feedAlgorithm = FeedAlgorithm.ReverseChronological
-      } else {
+      if (algorithm && algorithm !== FeedAlgorithm.ReverseChronological) {
         throw new InvalidRequestError(`Unsupported algorithm: ${algorithm}`)
       }
 
-      let postsQb = db
-        .selectFrom('post')
-        .select([
-          sql<FeedItemType>`${'post'}`.as('type'),
-          'uri as postUri',
-          'cid as postCid',
-          'creator as originatorDid',
-          'indexedAt as cursor',
-        ])
+      const feedService = ctx.services.feed(ctx.db, ctx.imgUriBuilder)
```

Comment:
👍 nice to see the new service parameter fitting in without too much trouble.

Reply:
yea really slick 👍